src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -39,10 +40,13 @@ public class AuthorizationResolver : IAuthorizationResolver
 
     public AuthorizationResolver(
         RuntimeConfigProvider runtimeConfigProvider,
-        IMetadataProviderFactory metadataProviderFactory
-        )
+        IMetadataProviderFactory metadataProviderFactory,
+        HotReloadEventHandler<HotReloadEventArgs>? handler = null)
```

Comment:
remind me is this handler filled in by Dependency injection?

Reply:
Yes. We register that in startup: 
https://github.com/Azure/data-api-builder/blob/c9aa87b0257507b1ea223953cfc8b1c76928610b/src/Service/Startup.cs#L81-L82

`HotReloadEventHandler<HotReloadEventArgs>?` is nullable to avoid cascading changes within the 7 other test files that don't mock the hotreloadeventhandler. When nullable, `hander?.Subscribe()` is a no-op. The tests included in this PR add coverage for supplying a hotreloadeventhandler for hot-reloading.